publish.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# This script can be executed to publish all results found in `result` subdir. It will
+# compress the loader image because Capstan requires compressed file when pulling 
+# images from the repository. Then it will look for all package files and their manifests
+# (*.mpm and *.yaml) and upload them sequentially.
+#
+# Before using this script, you have to configure s3cmd (http://s3tools.org/s3cmd).
+# After installing, you can invoke the configuration command
+#
+# $ s3cmd --configure
+#
+# You will be asked few AWS related questions (region, key, secret) that you need to
+# provide for the user with write access.
+
+pushd result
+
+echo "Compressing OSv loader image"
+gzip osv-loader.qemu
+echo "Uploading OSv loader image"
+s3cmd put --acl-public --guess-mime-type osv-loader.qemu.gz s3://mikelangelo-capstan/mike/osv-loader/osv-loader.qemu.gz
+
+FILES=`ls -1 *.{yaml,mpm}`
+for f in $FILES; do
+    echo "Uploading package file: $f"
+    s3cmd put --acl-public --guess-mime-type $f s3://mikelangelo-capstan/packages/$f
```

Comment:
If it terminates, then yes. However, s3cmd will do its own retries. For example, this morning several packages were retried multiple times because of poor internet connection. s3cmd sets a short delay (that increases, 3, 6, 9, 12, ... seconds) and retries the transmission.